test/jdk/java/awt/print/PrinterJob/ValidatePage/ValidatePage.java

```diff
@@ -21,407 +21,316 @@
+        epp.setLayout(new GridLayout(6, 2, 0, 0));
+
+        epp.add(new Label("Page width:"));
+        epp.add(tpw = new TextField());
+        epp.add(new Label("Page height:"));
+        epp.add(tph = new TextField());
+        epp.add(new Label("Left Margin:"));
+        epp.add(tpglm = new TextField());
+        epp.add(new Label("Top margin:"));
+        epp.add(tpgtm = new TextField());
+        epp.add(new Label("Imageable Wid:"));
+        epp.add(tpgiw = new TextField());
+        epp.add(new Label("Imageable Hgt:"));
+        epp.add(tpgih = new TextField());
+
+        add(epp);
+        displayPageFormatAttributes();
+
+        Panel panel = new Panel();
+        Button defButton = new Button("Default Page");
+        defButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                myPageFormat = myPrinterJob.defaultPage();
+                displayPageFormatAttributes();
+            }
+        });
+
+        Button pageButton = new Button("Page Setup..");
+        pageButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                myPageFormat = myPrinterJob.pageDialog(myPageFormat);
+                displayPageFormatAttributes();
+            }
+        });
+        Button printButton = new Button("Print");
+        printButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                try {
+                    //if (myPrinterJob.printDialog()) {
+                    myPrinterJob.setPrintable(ValidatePage.this,
+                            myPageFormat);
+                    myPrinterJob.print();
+                    // }
+                } catch (PrinterException pe) {
+                    pe.printStackTrace();
                 }
-    });
-    Button setButton = new Button ("Set Paper");
-    setButton.addActionListener(new ActionListener() {
-                public void actionPerformed (ActionEvent e) {
-                  try {
-                      Paper p = new Paper();
-                      double pwid = Double.parseDouble(tpw.getText());
-                      double phgt = Double.parseDouble(tph.getText());
-                      double pimx = Double.parseDouble(tpglm.getText());
-                      double pimy = Double.parseDouble(tpgtm.getText());
-                      double pimwid = Double.parseDouble(tpgiw.getText());
-                      double pimhgt = Double.parseDouble(tpgih.getText());
-                      p.setSize(pwid, phgt);
-                      p.setImageableArea(pimx, pimy, pimwid, pimhgt);
-                      myPageFormat.setPaper(p);
-                            displayPageFormatAttributes();
-                  } catch (NumberFormatException nfe) {
-                  }
+            }
+        });
+
+        Button chooseButton = new Button("Printer..");
+        chooseButton.addActionListener(new ActionListener() {
```

Comment:
Same changes at other places as well 

```suggestion
        chooseButton.addActionListener(e -> {
```